src/main/java/org/prebid/server/cache/CacheService.java

```diff
@@ -350,24 +367,20 @@ private BidCacheResult createBidCacheResult(CacheObject cacheObject) {
         return result;
     }
 
-    public String getEndpointUrl() {
-        return endpointUrl;
-    }
-
     /**
      * Composes cached asset URL for the given UUID cache value.
      */
     public String getCachedAssetURL(String uuid) {
-        return cachedAssetUrlTemplate.replaceFirst("%PBS_CACHE_UUID%", uuid);
+        return cachedAssetUrlTemplate.toString().replaceFirst("%PBS_CACHE_UUID%", uuid);
     }
 
     /**
      * Composes prebid cache service url against the given schema and host.
      */
-    public static String getCacheEndpointUrl(String cacheSchema, String cacheHost) {
+    public static URL getCacheEndpointUrl(String cacheSchema, String cacheHost, String path) {
         try {
             final URL baseUrl = getCacheBaseUrl(cacheSchema, cacheHost);
-            return new URL(baseUrl, "/cache").toString();
+            return new URL(baseUrl, "/" + path);
```

Comment:
1. Let's assume the `path` config property will be with leading slash:
```
cache.path=/cache
```
2. We don't need to concat leading slash when building URL.